eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/source/protocol/impl/CloudEventProtocol.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.connector.http.source.protocol.impl;
+
+import org.apache.eventmesh.connector.http.common.BoundedConcurrentQueue;
+import org.apache.eventmesh.connector.http.source.config.SourceConnectorConfig;
+import org.apache.eventmesh.connector.http.source.protocol.Protocol;
+import org.apache.eventmesh.openconnect.offsetmgmt.api.data.ConnectRecord;
+import org.apache.eventmesh.openconnect.util.CloudEventUtil;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.http.vertx.VertxMessageFactory;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.Route;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * CloudEvent Protocol.
+ */
+@Slf4j
+public class CloudEventProtocol implements Protocol {
+
+    // Protocol name
+    public static final String PROTOCOL_NAME = "CloudEvent";
```

Comment:
What is the difference between `CloudEventProtocol` and `CommonProtocol`? CloudEvent is the data exchange format within EventMesh cluster. Messages received by the source connector are converted to cloudevent and sent to the sink connector.

Reply:
The code in CloudEvent Protocol is from before, I just migrated it and encapsulated it non-destructively.
The most essential difference between the two is that CloudEventProtocol processes the request data and converts it into a CloudEvent object, while CommonProtocol only saves the request data without doing any processing on it.

Reply:
I don't see the need to convert the message to CloudEvent in the Source connector's processing logic, as it would be converted again in the EventMesh cluster, which would result in two nested duplicate CloudEvent data structures. The Webhook client does not need to support CloudEvent to work properly with the Webhook source connector, even supporting CloudEvent doesn't really bring any benefit. So I think it's fine to keep CommonProtocol as the default option to avoid ambiguity and remove/deprecate CloudEventProtocol.

Reply:
Since the current stable version only allows to receive CloudEvent protocol, if it is directly deleted, it will directly lead to version compatibility issues. If CloudEvent protocol is indeed not needed, I think it can be prompted by using `@Deprecated` first and changing the default protocol to CommonProtocol and completely deleted in future version.

Reply:
Conversation resolved by https://github.com/apache/eventmesh/pull/4913#issuecomment-2191773641.